### ISSUE TITLE
Add a value to `recruitment_pattern=` that signifies "any unit"

### ISSUE THREAD

**Issue by cooljeanius** [wesnoth/wesnoth#6006]
I was discussing this with @CelticMinstrel on Discord:
<img width="930" alt="Screen Shot 2021-08-03 at 09 21 29" src="https://user-images.githubusercontent.com/1388660/128022575-379e6cd9-a2ce-4cb1-99e9-76cc94e33119.png">
We can bikeshed about whether it should be called "`anything`" or "`random`" or something else, but the point is that there should be a way to add a slot to recruitment patterns for units that you don't care about what you get.

cc-ing @Elvish-Hunter because there might be `wmllint` changes that need to go with this.

**Comment by CelticMinstrel** [comment]
Personally, I think something symbolic (probably `*`) would be best, to better distinguish it from unit type IDs and usage strings.

**Comment by stevecotton** [comment]
According to the Wiki docs, it already exists and is the empty string; at least that's my interpretation of the docs for `[facet][value][recruit][type]=` in https://wiki.wesnoth.org/AI_Recruitment#Aspect_recruitment_instructions .

Have you tried `recruitment_pattern="fighter,archer,"` with the trailing comma?

**Comment by CelticMinstrel** [comment]
An empty string is a pretty terrible way to indicate that though…

**Comment by cooljeanius** [comment]
> Huh, I looked at that page but missed that sentence. An empty string is a pretty terrible way to indicate that though…

Yeah agreed; updating it to something better would be preferred

**Comment by cooljeanius** [comment]
33fbc52 helps for the level number part of this issue

**Comment by Wedge009** [comment]
> ...there might be `wmllint` changes that need to go with this.

I'm not sure what the recruitment pattern is about but there are current warnings from `wmllint` against EI S12, and TRoW S4a, 17a, 17b that seems to relate to this.

**Comment by CelticMinstrel** [comment]
Does the trailing comma actually work? It's pretty common for the empty strings to be stripped out when splitting into a list on commas, so it wouldn't surprise me if that happens here too.

**Comment by doofus-01** [comment]
Why does it need to be a trailing comma?  Can't you use a double comma?

**Comment by CelticMinstrel** [comment]
> Why does it need to be a trailing comma? Can't you use a double comma?

Assuming order doesn't matter, a trailing comma or a double comma or a leading comma are all the same thing.

**Comment by doofus-01** [comment]
Yes, I asked because you said empty strings got stripped out; does it matter if empty string is mid-list vs end-of-list?

**Comment by CelticMinstrel** [comment]
It doesn't make any difference, no.
